google/cloud/storage/blob.py

```diff
@@ -2058,6 +2060,15 @@ def _do_upload(
                   **only** response in the multipart case and it will be the
                   **final** response in the resumable case.
         """
+        if if_metageneration_match is None and num_retries is None:
```

Comment:
I think we should update the docstring as well to make it clear that setting num_retries can be used to override the idempotency requirement (assuming that's the intention), to make sure that users know the implications.

Reply:
Yes, that's the intention. Will do.